### PR TITLE
Convert `/navigation` folder to functional components

### PR DESCRIPTION
**Pull request by imjuangarcia**
This PR refactors the components under the `/navigation` folder, from class-based to functional ones.

### CODE REVIEW

File: components/navigation/navItem.js [hunk 1]
```diff
@@ -1,89 +1,87 @@
...
+  offScreen,
+  slug,
+  condensed,
+  paths,
+  version,
+  maxVersion,
+}) => {
+  const trueName = () => {
+    return page.path.split("/").pop();
+  };
```
Comment (tvst) [inline components/navigation/navItem.js:16]: `trueName` could just be a constant, no?

```
const trueName = page.path.split("/").pop();
```

File: components/navigation/navItem.js [hunk 2]
```diff
@@ -1,89 +1,87 @@
...
-    const props = this.props;
+  const cleanName = () => {
+    return trueName().replaceAll("-", " ");
+  };
 
-    let subNav;
-    let url_in_child = false;
+  const shouldHaveID = () => {
+    return offScreen ? page.name : false;
+  };
```
Comment (tvst) [inline components/navigation/navItem.js:24]: `shouldHaveId` could just be a constant, no?

```
const shouldHaveId = offScreen ? page.name : false;
```

File: components/navigation/sideBar.js [hunk 3]
```diff
@@ -1,110 +1,96 @@
...
+  const [state, setState] = useState({
+    condensed: false,
+    loading: true,
+    depth: 1,
+    sticky: false,
+    over: false,
+    open: false,
+    theme: "light-mode",
+    menu,
+  });
```
Comment (tvst) [inline components/navigation/sideBar.js:15]: I think it's more correct to pull each of these into their own `useState`:

```
const [isCondensed, setIsCondensed] = useState(false);
const [isLoading, setIsLoading] = useState(false);
...
```

**Review by tvst — APPROVED** [review]

File: components/navigation/navItem.js (hunk 1 above)
Comment (imjuangarcia) [reply components/navigation/navItem.js]: In fact, these weren't being used, so I deleted them on 51f922b

File: components/navigation/navItem.js (hunk 2 above)
Comment (imjuangarcia) [reply components/navigation/navItem.js]: In fact, these weren't being used, so I deleted them on 51f922b

File: components/navigation/sideBar.js (hunk 3 above)
Comment (imjuangarcia) [reply components/navigation/sideBar.js]: Makes sense! Added them on b18db14

**Review by tvst — APPROVED** [review]

**Pull request merged**